### PR TITLE
GitHub mock

### PR DESCRIPTION
**Pull request by nickmarston**
Creates classes and sample test for testing the github bug filer.

### CODE REVIEW

File: src/WorkItems/IGitHubConnection.cs [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+    /// <summary>
+    /// This interface allows for mocking of the low-level Gitjub connection class.
+    /// </summary>
+    internal interface IGitHubConnection
+    {
+        /// <summary>
+        /// Provide for the instantiation of the GitHub connection instance.
+        /// </summary>
+        /// <param name="organization">The AzureDevOps account URI for the connection.</param>
```
Comment (michaelcfanning) [inline src/WorkItems/IGitHubConnection.cs:18]: Update comment for GitHub. :)

File: src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemFilerTests.cs [hunk 2]
```diff
@@ -111,13 +129,16 @@ private void TestWorkItemFiler(SarifLog sarifLog, SarifWorkItemContext context)
...
 
-            var vssConnectionMock = new Mock<IVssConnection>();
-            vssConnectionMock
-                .Setup(x => x.ConnectAsync(It.IsAny<Uri>(), It.IsAny<string>()))
-                .Returns(Task.CompletedTask)
-                .Callback<Uri, string>(
-                    (uri, pat) =>
-                    {
+            FilingClient filingClient;
+            if (adoClient == true)
```
Comment (michaelcfanning) [inline src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemFilerTests.cs:133]: >adoClient == true [](start = 16, length = 17)

Style nit, don't compare to bollean literal, just is if (blah). While I'm here, please factor the branches of this conditional into two distinct helpers, for best readability and factoring. Nice work on this change!

File: src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemFilerTests.cs [hunk 3]
```diff
@@ -167,7 +228,15 @@ private void TestWorkItemFiler(SarifLog sarifLog, SarifWorkItemContext context)
             int expectedWorkItemsCount = context.GetProperty(ExpectedWorkItemsCount);
 
             createWorkItemCalled.Should().Be(expectedWorkItemsCount);
-            createAttachmentCalled.Should().Be(expectedWorkItemsCount);
+
+            if (adoClient == true)
```
Comment (michaelcfanning) [inline src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemFilerTests.cs:232]: >if (adoClient == true) [](start = 12, length = 22)

suggest reducing this to a single statement:

createAttachmentCount.Should().Be(adoClient ? expectedWorkItemsCount : 0);

Comment (nickmarston) [reply src/Test.UnitTests.Sarif.WorkItems/SarifWorkItemFilerTests.cs:232]: Oh, I really like this.

---
In reply to: [409067203](https://github.com/microsoft/sarif-sdk/pull/1852#discussion_r409067203) [](ancestors = 409067203)

**Pull request merged**